app/forms/journeys/additional_payments_for_teaching/correct_school_form.rb

```diff
@@ -0,0 +1,43 @@
+module Journeys
+  module AdditionalPaymentsForTeaching
+    class CorrectSchoolForm < Form
+      attribute :current_school_id
+      attribute :school_somewhere_else
+
+      delegate :address, :name, to: :current_school, prefix: true, allow_nil: true
+      delegate :eligibility, to: :claim
+      delegate :school_somewhere_else?, to: :eligibility
+
+      def save
+        claim.update(
+          eligibility_attributes: eligibility_attributes.with_indifferent_access.merge(
+            permitted_params
+          )
+        )
+        claim.reset_eligibility_dependent_answers(["current_school_id"])
+        true
+      end
+
+      def current_school_id
+        @current_school_id || current_school&.id
+      end
+
+      private
+
+      def current_school
+        @current_school ||= (claim.recent_tps_school || claim.school)
+      end
+
+      def change_school?
+        params[:change_school] == "true" || 
```

Comment:
Non blocker but should `change_school` be an attribute on this form? `attribute :change_school, :boolean` - would handle type casting for us and we wouldn't have to access the non permitted params